docs(package-plugin): clarify tool lookup behavior (#10335)
apple/swift-package-manager: Sources/Runtimes/PackagePlugin/Documentation.docc/Curation/PluginContext.md

# ``PackagePlugin/PluginContext``

Use a plugin context to inspect the package and locate the command-line tools
that a plugin needs.

> ## Locating Command-Line Tools

Reviewer: ```suggestion
## Locating command-line tools
```

(We want to keep to sentence casing for section headers)

Call ``tool(named:)`` with a tool's logical name to obtain the URL of an

> executable that runs on the host platform. SwiftPM builds executable dependencies

Reviewer: ```suggestion
executable that runs on the host platform. Swift Package Manager builds executable dependencies
```

for the host platform and selects host-compatible variants from binary artifact
bundles.

The lookup name depends on how the plugin declares the tool:

| Dependency | Lookup name |
| --- | --- |
| Executable target in the same package | Target name |
| Executable product from another package | Product name |
| Executable in a binary target | Artifact name in the artifact bundle metadata |

Only direct dependencies of the plugin target provide tools. A declared tool
takes precedence over any executable with the same name in a host-provided
search directory. If a binary target declares the requested artifact but has no
variant for the host platform, lookup reports
``PluginContextError/toolNotSupportedOnTargetPlatform(name:)`` instead of
searching for another executable with that name.

If no declared dependency matches, ``tool(named:)`` checks each host-provided
search directory in order and returns the first executable file with the
requested name. On Windows, it appends the `.exe` suffix when searching these
directories. The directories are an implementation detail of the host and can

> differ between SwiftPM, an IDE, and other package-manager integrations. Declare

Reviewer: ```suggestion
differ between Swift Package Manager, an IDE, and other package-manager integrations. Declare
```
